src/http/chunk_parsers.rl

```diff
@@ -154,25 +154,16 @@ action mark {
     g.mark_start(p);
 }
 
-action move {
-    g.mark_end(p);
-    g.mark_start(p);
-}
-
 action store_field_name {
     _field_name = str();
 }
 
-action no_mark_store_value {
-    _value = get_str();
+action trim_trailing_whitespace_and_store_value {
+    _value = str();
+    trim_trailing_spaces_and_tabs(_value);
```

Comment:
It's a little cleaner to trim before the call to str(). However, in practice it doesn't make a difference since trailing whitespace will be rare (much rarer than intermediate whitespace).

Reply:
Before the call to str(), the trailing whitespace might be split between what's already in the internal "checkpoint" buffer of `sstring_builder`, and what's in the parsed buffer and not copied yet. (This happens iff a buffer ends inside the trailing whitespace — you need to handle that case too). So it's hardly cleaner — just more hoops to jump through.
Unless you mean that it's cleaner in the sense that it produces a cleaner result — it doesn't leave unused capacity. Then you are right — but it also shouldn't matter, as you noted.

Reply:
sstring::resize() reallocates (sstring doesn't have separate capacity/size).

Reply:
I mean, it does reallocate when you resize up, but it doesn't when you resize down.

Reply:
Ah. Then it's even more minor.